sw/device/lib/base/log.c

```diff
@@ -64,11 +50,20 @@ void base_log_internal_core(log_severity_t severity, const char *file_name,
  * Logs |format| and the values that following in an efficient, DV-testbench
  * specific way.
  *
- * @param severity the log severity.
+ * @param nargs the number of variable parameters passed to the format string.
+ * @param log_ptr pointer (address) of the log string.
  * @param format a format string, as described in print.h. This must be a string
  * literal.
  * @param ... format parameters matching the format string.
  */
-void base_log_internal_dv(log_severity_t severity, const char *format, ...) {
-  // Do nothing, for now.
-}
\ No newline at end of file
+void base_log_internal_dv(uint32_t nargs, uintptr_t log_ptr, const char *format, ...) {
```

Comment:
Since you want to log the filename, the line number, and the severity, I'm not sure why you can't take those as separate arguments here and write them to the backdoor, too, before writing the arguments.

In fact, I think you can have the exact same signature as base_log_internal_core.

Reply:
I have changed the implementation by writing the address to the struct that holds these values. Though, the rationale is to get to absolute minimum processing (want to bring the logging down to just a few instructions) at run time by pushing all the processing to the script.